.github/workflows/release.yaml

```diff
@@ -75,55 +60,39 @@ jobs:
       - name: Fetch git tags
         run: git fetch --tags --force
 
-      # Configure git user name/email for creating annotated version tag.
-      - name: Setup git config
-        run: |
-          git config user.name "Coder CI"
-          git config user.email "dean+cdrci@coder.com"
-
-      - name: Create release tag and release notes
+      - name: Create release notes
+        if: ${{ inputs.dry_run }}
+        env:
+          # We always have to set this since there might be commits on
+          # main that didn't have a PR.
+          CODER_IGNORE_MISSING_COMMIT_METADATA: "1"
         run: |
           set -euo pipefail
           ref=HEAD
           old_version="$(git describe --abbrev=0 "$ref^1")"
 
-          if [[ "${{ inputs.ignore_missing_commit_metadata }}" == *t* ]]; then
-            export CODER_IGNORE_MISSING_COMMIT_METADATA=1
-          fi
-
-          # Warn if CODER_IGNORE_MISSING_COMMIT_METADATA is set any other way
-          # than via dry-run.
-          if [[ ${CODER_IGNORE_MISSING_COMMIT_METADATA:-0} != 0 ]]; then
-            echo "WARNING: CODER_IGNORE_MISSING_COMMIT_METADATA is enabled and we will ignore missing commit metadata." 1>&2
-          fi
-
-          version_args=()
-          if [[ $CODER_DRY_RUN == *t* ]]; then
-            # Allow dry-run of branches to pass.
-            export CODER_IGNORE_MISSING_COMMIT_METADATA=1
-            version_args+=(--dry-run)
-          fi
-
           # Cache commit metadata.
           . ./scripts/release/check_commit_metadata.sh "$old_version" "$ref"
 
-          declare -p version_args
-
-          # Create new release tag (note that this tag is not pushed before
-          # release.sh is run).
-          version="$(
-            ./scripts/release/tag_version.sh \
-              "${version_args[@]}" \
-              --ref "$ref" \
-              --"$CODER_RELEASE_INCREMENT"
-          )"
+          if [[ $CODER_DRY_RUN != *t* ]]; then
+            # This will fail if we're doing a non-dry-run without tag
+            # being present.
+            version="$(./scripts/version.sh)"
+          else
+            # Create a tag for this dry-run (local only, can't be uploaded).
+            version="$(
+              ./scripts/release/tag_version.sh \
+                --ref "$ref" \
+                --"$CODER_RELEASE_INCREMENT"
+            )"
```

Comment:
How about we make it so the version comes from a workflow_dispatch input to simulate similar behavior to pushing a tag?

Reply:
Sure, sounds good!

Reply:
On second thought, is it perhaps best to keep as-is to mimic `release.sh` here, since that's how releases are supposed to be made?

Reply:
I think it would be best to mimic how the workflow gets started typically. If we wanted to test the version generating stuff, we can test release.sh too with the dry run flag.

Reply:
I thought about it one more time, and since we generate binaries with a version... we probably shouldn't modify the generated version anyway. Calling them vNEXT-devel. can only lead to confusion if they're ever used. So I just removed the whole increment input and simplified the step. 😅